src/cpp/utils/SystemInfo.cpp

```diff
@@ -83,4 +86,76 @@ bool SystemInfo::file_exists(
     return (stat(filename.c_str(), &s) == 0 && s.st_mode & S_IFREG);
 }
 
+ReturnCode_t SystemInfo::load_environment_file(
+        const std::string& filename,
+        const std::string& env_name,
+        std::string& env_value)
+{
+    // Check that the file exists
+    if (!SystemInfo::file_exists(filename))
+    {
+        return ReturnCode_t::RETCODE_BAD_PARAMETER;
+    }
+
+    // Read json file
+    std::ifstream file(filename);
+    nlohmann::json file_content;
+
+    try
+    {
+        file >> file_content;
+    }
+    catch (const nlohmann::json::exception&)
+    {
+        return ReturnCode_t::RETCODE_ERROR;
+    }
+
+    try
+    {
+        env_value = file_content.at(env_name);
+        if (env_value.empty() || env_value.compare("") == 0)
+        {
+            return ReturnCode_t::RETCODE_NO_DATA;
+        }
+    }
+    catch (const nlohmann::json::exception&)
+    {
+        return ReturnCode_t::RETCODE_NO_DATA;
+    }
+    return ReturnCode_t::RETCODE_OK;
+}
+
+FileWatchHandle SystemInfo::watch_file(
+        std::string filename,
+        std::function<void(const std::string&)> callback)
+{
+#if defined(_WIN32) || defined(__unix__)
```

Comment:
Even though usually Fast DDS excludes MacOS using `ifndef __APPLE__`, I think it is better this way because there are users employing other platforms that could have its own API.